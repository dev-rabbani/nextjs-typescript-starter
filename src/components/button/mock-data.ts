import {
  FillButtonProps,
  FillRedirectButtonProps,
  OutlineButtonProps,
  OutlineRedirectButtonProps,
} from './interface';

export const buttonProps: FillButtonProps = {
  label: 'Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-danger-text-white',
  size: 'lg',
  clickHandler: () => {},
  variant: 'fill',
};

export const redirectButtonProps: FillRedirectButtonProps = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-danger-text-white',
  size: 'lg',
  variant: 'fill',
};

export const outlineButtonProps: OutlineButtonProps = {
  label: 'Outline Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-warning',
  size: 'lg',
  clickHandler: () => {},
  variant: 'outline',
};

export const redirectOutlineButtonProps: OutlineRedirectButtonProps = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-warning',
  size: 'lg',
  variant: 'outline',
};
