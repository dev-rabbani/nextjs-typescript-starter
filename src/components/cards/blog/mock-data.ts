import { BlogCardProps } from './interface';

export const blogCardProps: BlogCardProps = {
  slug: 'https://example.com/',
  image: {
    src: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
    alt: 'dummy image 2',
  },
  title: 'Headless Website Development',
};
