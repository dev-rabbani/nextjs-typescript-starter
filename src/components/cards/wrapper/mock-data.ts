import { CardWrapperProps } from './interface';

export const cardWrapperProps: CardWrapperProps = {
  cardBorderRadius: 'radius-none',
  cardShadow: 'shadow-one',
};
