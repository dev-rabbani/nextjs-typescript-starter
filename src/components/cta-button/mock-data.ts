import { ICtaButton } from './interface';

export const ctaButtonProps: ICtaButton = {
  label: 'Call to action',
  href: '/',
  colorScheme: 'primary',
  isOpenNewTab: false,
};
