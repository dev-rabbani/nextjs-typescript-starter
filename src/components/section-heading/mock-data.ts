import { ISectionHeading } from './interface';

export const sectionHeadingProps: ISectionHeading = {
  subtitle: 'Sub Title',
  title: 'What Makes Us Different?',
  description:
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Dolorem est fugiat ea voluptatem rerum unde necessitatibus, eos ex maxime laborum?',
  colorScheme: 'primary',
  hasBottomSpacing: true,
  isCenter: false,
};
