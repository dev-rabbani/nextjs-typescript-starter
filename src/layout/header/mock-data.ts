import { IHeader } from './interface';

export const headerProps: IHeader = {
  logo: 'Starter',
  menuItems: [
    {
      label: 'Home',
      href: '/',
    },
    {
      label: 'About',
      href: '/',
    },
    {
      label: 'Service',
      href: '/',
    },
    {
      label: 'Blog',
      href: '/',
    },
  ],
};
