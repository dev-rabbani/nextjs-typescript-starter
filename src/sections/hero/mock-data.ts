import { IHero } from './interface';

export const heroProps: IHero = {
  title:
    'Welcome to Nextjs typescript starter, the ultimate solution for building modern web applications with Next.js!',
  description:
    'Our starter template is designed to give you a head start on your project, with all the tools you need to create a beautiful and responsive web application in no time. With Next.js, you can enjoy the benefits of server-side rendering and a streamlined development process, while still having the flexibility to use the front-end framework of your choice',
  buttons: {
    buttonLeft: {
      label: 'Server Side Example',
      href: '/',
    },
    buttonRight: {
      label: 'Client Side Example',
      href: 'client-side',
    },
  },
};
