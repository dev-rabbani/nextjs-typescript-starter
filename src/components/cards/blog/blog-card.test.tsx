import { render } from '@testing-library/react';
import BlogCard from '.';
import { blogCardProps } from './mock-data';

it('renders course card correctly', () => {
  const { getAllByRole } = render(<BlogCard {...blogCardProps} />);
  const element = getAllByRole('presentation')[0];
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
