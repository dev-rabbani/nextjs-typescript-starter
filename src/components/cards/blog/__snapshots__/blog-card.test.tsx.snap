// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders course card correctly 1`] = `
<div
  class="relative overflow-hidden h-auto card shadow-one radius-one"
  role="presentation"
>
  <a
    class="block"
    href="https://example.com/"
  >
    <div
      class="group/card relative h-full bg-success/[.2]"
    >
      <div
        class="overflow-hidden"
      >
        <img
          alt="dummy image 2"
          class="transition-transform ease-in-out group-hover/card:scale-110"
          data-nimg="1"
          decoding="async"
          height="720"
          loading="lazy"
          src="http://localhost/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1920&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1080&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1920&q=75 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="960"
        />
      </div>
      <div
        class="flex flex-col space-y-3 p-5 pt-8"
      >
        <h3
          class="line-clamp-2 flex-1 text-lg font-bold"
        >
          Headless Website Development
        </h3>
      </div>
    </div>
  </a>
</div>
`;
